src/test/java/org/apache/sling/resourceresolver/impl/mapping/VanityPathMapEntriesTest.java

```diff
@@ -149,6 +154,36 @@ public void setup() throws Exception {
         Optional<ResourceResolverMetrics> metrics = Optional.empty();
 
         mapEntries = new MapEntries(resourceResolverFactory, bundleContext, eventAdmin, stringInterpolationProvider, metrics);
+        waitForBgInit();
+    }
+
+    // get internal flag that signals completion of background task
+    private AtomicBoolean getVanityPathsProcessed() {
+        try {
+            Field field = MapEntries.class.getDeclaredField("vanityPathsProcessed");
+            field.setAccessible(true);
+            return (AtomicBoolean) field.get(mapEntries);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    // wait for background thread to complete
+    private void waitForBgInit() {
+        while (!getVanityPathsProcessed().get()) {
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                // ignored
+            }
+        }
+    }
+
+    // get vanity paths (after waiting for bg init to complete)
+    private void initializeVanityPaths() throws IOException {
+        getVanityPathsProcessed().set(false);
+        mapEntries.initializeVanityPaths();
+        waitForBgInit();
```

Comment:
do I get it right, that you added the ``waitForBgInit()`` just in preparation for the next change, where you will run tests also with ``isVanityPathCacheInitInBackground = true``?

Reply:
The tests (now) run with bginit as well, see parameters of the test class.